appinsights/clock_test.go

```diff
@@ -0,0 +1,34 @@
+package appinsights
+
+import (
+	"time"
+
+	"code.cloudfoundry.org/clock"
+	"code.cloudfoundry.org/clock/fakeclock"
+)
+
+var fakeClock *fakeclock.FakeClock
+
+func mockClock() {
+	fakeClock = fakeclock.NewFakeClock(time.Now().Round(time.Minute))
+	currentClock = fakeClock
+}
+
+func resetClock() {
+	fakeClock = nil
+	currentClock = clock.NewClock()
+}
+
+func slowTick(seconds int) {
+	const delay = time.Millisecond * time.Duration(5)
+
+	// Sleeps in tests are evil, but with all the async nonsense going
```

Comment:
Test code.  Think `sinon.clock.tick()` except we have no way to know whether the worker queue is empty or debounce the time advancement.  We just have to wait for things to finish.  It seems to work.